vllm/engine/arg_utils.py

```diff
@@ -49,6 +49,18 @@
 TypeHintT = Union[type[T], object]
 
 
+class DeprecatedAction(argparse.Action):
```

Comment:
Let's move this out of a PR. I also have another PR that address this

Reply:
ok, I will use the new approach after your PR #17426 is merged.